### ISSUE TITLE
loss-->nan

### ISSUE THREAD

**Issue by mowangmodi** [rover-xingyu/Ha-NeRF#25]
Hi，xingyu！
In the calculation of loss,
 ret ['mode_seeking ']=hparams. weightMS * 1/\
((torch.mean(torch.abs(inputs['rgb_fine'].detach() - inputs['rgb_fine_random'])) / \
Torch.mean (torch. abs (inputs ['a_embedded ']. detach() - inputs ['a_embedded_random']. detach())+1 * 1e-5),
 the denominator here is 0,（Because inputs ['a_embedded ']. detach() and inputs ['a_embedded_random'] are the same） and loss will result in nan. I don't quite understand, can you explain?

**Comment by rover-xingyu** [comment]
Mode seeking is a regularization method proposed in [MSGAN](https://github.com/HelenMao/MSGAN/) to address the mode collapse issue for GANs. It is weird to have zero in the denominator since we add 1 * 1e-5 after the absolute value.

**Comment by mowangmodi** [comment]
xingyu，The following are three different issues that occurred in the first three steps of my debugging, and I am very confused. I hope you can explain them to me. Thank you very much.

The following are the parameters I used, similar to the example given by GitHub:
--root_dir ./brandenburg_gate/ --dataset_name phototourism --save_dir save --img_downscale 2 --N_importance 64 --N_samples 64 --num_epochs 20 --batch_size 1024 --optimizer adam --lr 5e-4 --lr_scheduler cosine --exp_name exp_HaNeRF_Brandenburg_Gate --N_vocab 1500 --use_mask --encode_a --encode_random --num_gpus 1


- [ 1 ]  The first step is that Nan appears in ret ["mode_seeking"] in loss, which I also find very strange
- 
![1](https://user-images.githubusercontent.com/103345858/230533697-fc6fc74a-4940-45e9-830a-02c1d6857e0d.png)

- [ 2 ]In the second step of debugging, when I enter the fine network (where there is no nan in the Coarse network), I obtain a value of nan 
- 
![2](https://user-images.githubusercontent.com/103345858/230533907-8added8d-9173-4c70-8c3d-c75aea27d7e7.png)

- [ 3 ] In the second step of debugging, the randomly obtained images are fed into self. nc_ In the a (E_attr()) network, the output value is nan, but in the first two debugging steps, a 48 dimensional vector can be generated, so it is also strange here
- 
![3](https://user-images.githubusercontent.com/103345858/230534252-c559af4a-9aac-4984-9ada-4825a638b3af.png)

- [ 4 ] For the fourth point, the paper states that The appearance encoder E φ  Consists of 5 convolution layers, but there are 6 convolutional layers in the code, which I am very confused about
- 
![4](https://user-images.githubusercontent.com/103345858/230534554-01ce7e3f-0a40-4146-8d90-cbe62a816e4f.png)
The orange section is described in the paper 
![5](https://user-images.githubusercontent.com/103345858/230534671-e38fa08d-c06d-4fa7-919a-8508a5739ff8.png)

Overall, the above are the areas where I am very confused and I hope to receive your help. Thank you very much

**Comment by rover-xingyu** [comment]
As shown in the code, there are 5 convolution layers (from size to size/16), an adaptive average pooling (from size/16 to 1), and a fully-connected layer(from 1 to 1). I can not reproduce the problem of nan. Did you use the original Brandenburg_Gate dataset?

**Comment by mowangmodi** [comment]
yes, I use the original Brandenburg_Gate dataset.763 image training, 10 image testing.
![11](https://user-images.githubusercontent.com/103345858/230578803-77123600-e024-4dbb-8488-5a9ce8556d8c.png)

**Comment by rover-xingyu** [comment]
Maybe you could turn off the option of --encode_random and try again to see if the issue still exists.

**Comment by mowangmodi** [comment]
Thank you very much. I turn off the option of -- encode_ Random, this problem is solved. I turn off the option of -- encode_ Does the parameter 'random' have any impact on the results-- encode_ What is the purpose of the parameter 'random'?

**Comment by rover-xingyu** [comment]
It is used for view-consistent loss, as described in the paper. If you don care about it, you can eliminate this part.
